eppo_core/src/event_ingestion/event_delivery.rs

```diff
@@ -48,19 +60,27 @@ impl EventDelivery {
             .header("X-Eppo-Token", sdk_key.as_str())
             .json(&body)
             .send()
-            .await?;
+            .await
+            .map_err(EventDeliveryError::RetriableError)?;
         let response = response.error_for_status().map_err(|err| {
             return if err.status() == Some(StatusCode::UNAUTHORIZED) {
-                // TODO: Mark all events as failed, indicate that this error is not-retriable
+                // This error is not-retriable
                 log::warn!(target: "eppo", "client is not authorized. Check your API key");
-                Error::Unauthorized
+                EventDeliveryError::NonRetriableError(err)
+            } else if err.status() == Some(StatusCode::BAD_REQUEST) {
+                // This error is not-retriable
+                log::warn!(target: "eppo", "received 400 response delivering events: {:?}", err);
+                EventDeliveryError::NonRetriableError(err)
             } else {
-                // TODO: Mark all events as failed, indicate that this error is not-retriable
-                log::warn!(target: "eppo", "received non-200 response while fetching new configuration: {:?}", err);
-                Error::from(err)
+                // Other errors **might be** retriable
+                log::warn!(target: "eppo", "received non-200 response delivering events: {:?}", err);
+                EventDeliveryError::RetriableError(err)
```

Comment:
minor: I think we should do a bit more work distinguishing when we should retry or not. 4xx are usually non-retriable except 408 and 429. 5xx are usually retriable. (Marking this minor as retrying is a safe default.)

Reply:
👍🏽 i was debating whether or not to tackle this here. took a note for follow up in a future PR